lib/saml_idp/assertion_builder.rb

```diff
@@ -73,6 +77,81 @@ def fresh
     alias_method :raw, :fresh
     private :fresh
 
+    def encrypt(opts = {})
```

Comment:
Correct me if I'm wrong but the new code in this class doesn't much use the logic that existed previously and the existing code doesn't rely on this new code. So I was thinking a class onto its own that doesn't need to inerhit assertion builder at all, so in encrypt you'd call `EncryptedAssertionBuilder.new(opts).build`. Does that make sense?